samples/apps/tinyRA/tinyra/tui.py

```diff
@@ -0,0 +1,1186 @@
+    - ctrl+g: memorize the autogen message
+    """
+
+    BINDINGS = [
+        ("ctrl+c", "request_quit", "Quit"),
+        ("ctrl+s", "request_settings", "Settings"),
+    ]
+
+    CSS_PATH = "tui.css"
+
+    TITLE = "TinyRA"
+    SUB_TITLE = "A minimalistic, long-lived research assistant"
+
+    def compose(self) -> ComposeResult:
+        """Create child widgets for the app."""
+        yield Grid(
+            Header(show_clock=True),
+            Grid(
+                DirectoryTreeContainer(id="directory-tree"),
+                Grid(
+                    Button("Delete", variant="error", id="delete-file-button"),
+                    Button("Empty Work Dir", variant="error", id="empty-work-dir-button"),
+                    id="directory-tree-footer",
+                ),
+                id="directory-tree-grid",
+            ),
+            Container(
+                ChatDisplay(id="chat-history"),
+                ChatInput(id="chat-input"),
+                id="chat-container",
+            ),
+            Footer(),
+            id="main-grid",
+        )
+
+    def action_request_quit(self) -> None:
+        self.push_screen(QuitScreen())
+
+    def action_toggle_dark(self) -> None:
+        """An action to toggle dark mode."""
+        self.dark = not self.dark
+
+    def action_request_settings(self) -> None:
+        self.push_screen(SettingsScreen())
+
+    def on_button_pressed(self, event: Button.Pressed) -> None:
+        if event.button.id == "empty-work-dir-button":
+            # loop through each file in the work dir and delete it
+            work_dir = os.path.join(APP_CONFIG.get_data_path(), "work_dir")
+            for file in os.listdir(work_dir):
+                file_path = os.path.join(work_dir, file)
+                if os.path.isfile(file_path):
+                    os.remove(file_path)
+                elif os.path.isdir(file_path):
+                    shutil.rmtree(file_path)
+        elif event.button.id == "delete-file-button":
+            dir_tree = self.query_one("#directory-tree > DirectoryTree", DirectoryTree)
+            highlighted_node = dir_tree.cursor_node
+
+            if highlighted_node is not None:
+                dir_tree.action_cursor_up()
+                file_path = str(highlighted_node.data.path)
+
+                APP_CONFIG.delete_file_or_dir(file_path)
+
+    def on_directory_tree_file_selected(self, event: DirectoryTree.FileSelected) -> None:
+        """Called when the user click a file in the directory tree."""
+        event.stop()
+        try:
+            # open the file using the default app
+            logging.info(f"Opening file {event.path}")
+            # check if the app is running in a codespace
+            if os.environ.get("CODESPACES"):
+                os.system(f"code {event.path}")
+            else:
+                # open the file using the default app
+                os.system(f"open '{event.path}'")
+        except Exception:
```

Comment:
In my wsl ubuntu 20 install 'xdg-open' is needed instead of 'open'
os.system(f"xdg-open '{event.path}'")

https://stackoverflow.com/questions/264395/linux-equivalent-of-the-mac-os-x-open-command

I have not tried windows but this was suggested for windows:
os.system(f"start '' '{event.path}'")